pkg/costmodel/key.go

```diff
@@ -75,9 +74,11 @@ func (k podKey) String() string {
 
 func newPodKey(cluster, namespace, pod string) podKey {
 	return podKey{
-		Cluster:   cluster,
-		Namespace: namespace,
-		Pod:       pod,
+		namespaceKey: namespaceKey{
+			Cluster:   cluster,
+			Namespace: namespace,
+		},
+		Pod: pod,
 	}
```

Comment:
Optimization here which seemed to have a fairly significant reduction in allocation upon very loose observation via pprof. The benefits from non-string keys are still very hard to pin down, as the keys are escaped at some point and count towards heap allocations. I did not expect embedding namespaceKey to have that dramatic of an effect, but it appears to have helped for `appendLabels()`

Reply:
I'll be honest, I'm not completely sold, but I did test this in isolation and there were noticeable changes. I still need to use some better benchmarks for measurement.